oss-internship-2020/curl/tests/callbacks.cc

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "callbacks.h"
+
+#include <cstdlib>
+#include <cstring>
+
+size_t write_to_memory(char* contents, size_t size, size_t nmemb, void* userp) {
+  size_t realsize = size * nmemb;
+  MemoryStruct* mem = (MemoryStruct*)userp;
```

Comment:
Do not use C-style casts. Use a C++ cast instead. You can also use `auto` to avoid repeating the type name:

```c++
auto* mem = static_cast<MemoryStruct*>(userp);
```

See https://google.github.io/styleguide/cppguide.html#Casting